src/core/toga/widgets/base.py

```diff
@@ -196,7 +205,8 @@ def window(self, window):
         self._set_window(window)
 
     def _set_window(self, window):
-        pass
+        if window is not None:
```

Comment:
This should be in the base setter, not the `_set_window()` call. This method only exists so that subclasses can extend the setter; there's no expectation that you'll need to call `super()._set_window()`.

Reply:
Done